crates/matrix-sdk/src/account.rs

```diff
@@ -1087,6 +1087,30 @@ impl Account {
         Ok(media_preview_config)
     }
 
+    /// Get the media preview configuration event content stored in the cache.
+    ///
+    /// This will return the default configuration if no value was found.
+    /// Will check first for the stable event and then for the unstable one.
+    pub async fn get_stored_media_preview_config_event_content(
```

Comment:
Could `observe_media_preview_config` use this method to get the initial value, or is there a reason that it must make a request to the homeserver?

Reply:
Well, historically that has been a bit of a footgun because the sliding sync proxy tended to be a bit slow to sync the account data.

So you would get a stale value. I suppose it's not that bad to get a stale value here since you're going to receive the update eventually.

In conclusion, yeah I think you're right.

Reply:
okay perfect then I'll rework the existing code to use this in place of the API call

Reply:
Another question should I also do the same on the setters?
https://github.com/matrix-org/matrix-rust-sdk/blob/0cfb4b1c55dd2ff8e90a6bb804944592ea819222/crates/matrix-sdk/src/account.rs#L1114-L1142

Or the setters should keep fetching the API directly?

Reply:
Answered in the chat room, but for completeness sake. The setters should still send out the request, in case the sync is late or not running, so we don't edit stale data.